### PR TITLE
Refactor components inside ChannelNotificationsModal (not yet migrated to redux)

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix incorrect channel notification settings when switching teams and refactor `ChannelNotificationsModal`
Note that I've created separate sections although they are almost the same or have the same pattern so that it can be easily modify in the future whenever necessary.

#### Ticket Link
Jira ticket: [PLT-PLT-8515](https://mattermost.atlassian.net/browse/PLT-8515)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Touches critical sections of the codebase (refactor `ChannelNotificationsModal`)

### CODE REVIEW

**Comment by hmhealey** [comment]
release-4.6 will be merged into master at some point, so maybe this should be based off [this PR](https://github.com/mattermost/mattermost-webapp/pull/560) to not conflict when that goes in?

**Comment by saturninoabril** [comment]
@hmhealey that commit from release-4.6 is now merged into master.
PR is now updated fixing merge conflicts.

File: components/channel_push_notification_section.jsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                    id='channel_notifications.never'
+                    defaultMessage='Never'
+                />
+            );
+        }
+
+        return levelName;
+    }
+
+    getDescribe(globalNotifyLevelName) {
```
Comment (jespino) [inline components/channel_push_notification_section.jsx]: I think this is duplicated code too [here](https://github.com/mattermost/mattermost-webapp/pull/561/files#diff-33ce27fd0efb4acf8686e180212c6a22R96).

File: components/channel_push_notification_section.jsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+                                id='channelNotificationAllActivity'
+                                type='radio'
+                                name='desktopNotificationLevel'
+                                value={NotificationLevels.ALL}
+                                checked={this.props.memberNotificationLevel === NotificationLevels.ALL}
+                                onChange={this.handleOnChange}
+                            />
+                            <FormattedMessage id='channel_notifications.allActivity'/>
+                        </label>
+                        <br/>
```
[the remark this answers is not in the document]
Comment (asaadmahmood) [reply components/channel_push_notification_section.jsx]: Yup, Ideally we should never use <br> tags for spacing as such, we have classes to give spacing at the top, for eg. `padding-top` or `margin-top` with further modifiers like `x2` etc, please have a look at `_modifiers.scss`. So we should use those instead.

File: components/channel_desktop_notification_section.jsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                                checked={this.props.memberNotificationLevel === NotificationLevels.NONE}
+                                onChange={this.handleOnChange}
+                            />
+                            <FormattedMessage id='channel_notifications.never'/>
+                        </label>
+                    </div>
+                </div>
+            )];
+
+            const extraInfo = (
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply components/channel_desktop_notification_section.jsx]: I've created a separate component instead to reuse for other section.

File: components/channel_desktop_notification_section.jsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+                />
+            );
+        }
+
+        const describe = this.getDescribe(globalNotifyLevelName);
+
+        return (
+            <SettingItemMin
+                title={sendDesktop}
+                describe={describe}
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply components/channel_desktop_notification_section.jsx]: Same here, separate component is now created.

File: components/channel_push_notification_section.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (saturninoabril) [reply components/channel_push_notification_section.jsx]: I just removed it. Using `padding-top`, makes the spacing larger and out of proportion.

**Comment by saturninoabril** [comment]
@jespino I moved it all to a new folder now, although the ChannelNotificationsModal is not yet migrated.
I like to have these smaller presentational components so those are easier to test.

File: components/channel_push_notification_section.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (saturninoabril) [reply components/channel_push_notification_section.jsx]: @asaadmahmood Could you give this a quick check if the spacing is still correct after I've deleted the `<br>`? Thanks!

[the remark this answers is not in the document]
Comment (saturninoabril) [reply components/channel_push_notification_section.jsx]: The spacing looks the same with or without the `<br>` but I could be wrong :)

screen shot 2018-01-26 at 8 40 51 pm

**Pull request merged**